poisson2d/modern.f90

```diff
@@ -0,0 +1,83 @@
+module rho_interface
+  !! Define a scalar function over 2-space.
+  implicit none
+
+  private
+  public :: rho, dp
+
+  integer, parameter :: precision = 15, range = 307
+  integer, parameter :: dp = selected_real_kind(precision, range)
+
+  interface 
+    pure real(dp) module function rho(x,y)
+      !! Poisson equation inhomogeneous term.
+      implicit none
```

Comment:
Just a question for my knowledge: why is there an `implicit none` here? Is the `implicit none` in the module not covering it?

Reply:
Yes, that is a common mistake that I also learned the hard way --- `implicit none` in a module propagates and applies to everything except `interface`, where you have to repeat it.... :(

Reply:
Thank you @certik for your answer. I will have to check all my codes ;)